src/components/Rockets.js

```diff
@@ -1,12 +1,32 @@
+/* eslint-disable no-unused-vars */
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import Rocket from './Rocketcomponent';
+import fetchrockets from '../utils/rockets_API';
+import { addrocks } from '../redux/rockets/rocks';
 
-const Rockets = () => (
-  <>
-    <Rocket imgsrc="dsfsdf" imgname="rock" rocketname="Falcon 1" rocketdescription="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 2 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth." />
-    <Rocket imgsrc="dsfsdf" imgname="rock" rocketname="Falcon 1" rocketdescription="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 2 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth." />
-    <Rocket imgsrc="dsfsdf" imgname="rock" rocketname="Falcon 1" rocketdescription="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 2 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth." />
-    <Rocket imgsrc="dsfsdf" imgname="rock" rocketname="Falcon 1" rocketdescription="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 2 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth." />
-  </>
-);
+const Rockets = () => {
+  const [rockets, setRockets] = useState([]);
+  const stat = useSelector((state) => state.rocksReducer);
+  const dipsatch = useDispatch();
+  const loadRockets = async () => {
+    const rocktObj = await fetchrockets();
+    if (stat.length !== rocktObj.length) dipsatch(addrocks(rocktObj));
+    setRockets(rocktObj);
+  };
```

Comment:
Every time I go to the page, it askes for a new api request, where it should be just one time. Please fix it.
![net](https://user-images.githubusercontent.com/72808808/132350809-8845e161-0480-482b-b085-1c7581b9a9f8.png)
Thank you

Reply:
yeah, but the dispatch method is only called once, I implemented it in this way,so if the API returns a new rocket entry it would be added to the page.

Reply:
The fetch makes the page reload slower, if you check the project requirements, the fetch should only happen once. The dispatch to save to store as well. So if you can please fix it.
You can put the conditional inside the use effect with the dispatch as well.
Thank you